SimG4CMS/Forward/src/TimingSD.cc

```diff
@@ -115,8 +115,8 @@ void TimingSD::getStepInfo(const G4Step* aStep) {
   const G4Track* newTrack = aStep->GetTrack();
   // exclude ECAL gflash spots inside MTD detectors,
   // which not possible correctly handle
-  primaryID = getTrackID(newTrack);
-  if (primaryID < -1) {
+  const G4String& rname = newTrack->GetVolume()->GetLogicalVolume()->GetRegion()->GetName();
+  if (isMTD() && rname == "EcalRegion") {
```

Comment:
I am not sure if lines 118 and 119 are needed and would propose all lines 116-119.

Reply:
ok, but at this point we do not even need ```isMTD()```